### DISCUSSION TITLE
flash content erase in bootloader region at run time

### DISCUSSION THREAD

**Discussion by Sathishkumar-rk** [zephyrproject-rtos/zephyr#53434] [Q&A] [answered]
Hi @danieldegrasse 

I'm facing issue when do flash erase before writing into bootloader initial location. After erase the location, when do read back that location it's returning previous content, the content is not erased from location. 

The above trial made on bootloader regioin [0x30000000 - 0x300000FF].

The primary and secondary slot location able to erase and swap content. Facing problem only on erasing part of bootloader region.

API used for erase flash content at run time, Is this correct API to erase the region?
**boot_erase_region()**
code snippet
```
   /* Write to the bootloader. */
    area_id = 0;
   
 rc = flash_area_open(area_id, &fap);
    if (rc != 0) {
        return BOOT_EFLASH;
    }
    boot_erase_region(fap,0x00,256);
```

IMXRT1160EVK Board using for this project and default FlexSPI NOR flash.

Thanks,
Sathishkumar K

**Comment by danieldegrasse** [comment, marked as answer]
Hi @Sathishkumar-rk, I would not use `boot_erase_region` in an application, since it appears to be part of the mcuboot private API. I would use `flash_area_erase` for this purpose. Either way, the erase size being used (256 bytes), is smaller than the minimum erase size supported by the is25wp128 (4Kbytes)

**Reply by Sathishkumar-rk** [reply]
Hi @danieldegrasse 

Thanks for the update, i can able to erase flash using 4kb size besides I'm erasing flash from mcuboot code only. Sorry, it's was not mentioned in earlier thread.

One more suggestions is required, after erasing flash I want to do the software reset from mcuboot code, is it possible to do that? If yes can you please point out the API for software reset.

I was tried to call NVIC_SystemReset() API, unfortunately getting compilation error like couldn't able to find reference for API in mcuboot. 

Thanks,
SathishKumar

**Reply by danieldegrasse** [reply]
I would look at `sys_reboot`: https://docs.zephyrproject.org/apidoc/latest/reboot_8h.html (available when `CONFIG_REBOOT=y`)

**Comment by Sathishkumar-rk** [comment]
Thanks for the update, I can able to utilize the sys_reboot in my code.
